lib/broadway_rabbitmq/producer.ex

```diff
@@ -137,12 +142,14 @@ defmodule BroadwayRabbitMQ.Producer do
   end
 
   def handle_info({:basic_deliver, payload, meta}, state) do
-    %{channel: channel, client: client} = state
-    %{delivery_tag: tag} = meta
+    %{channel: channel, client: client, config: config} = state
+    %{delivery_tag: tag, redelivered: redelivered} = meta
 
     ack_data = %{
       delivery_tag: tag,
-      client: client
+      client: client,
+      redelivered: redelivered,
+      requeue: config[:requeue]
```

Comment:
You can compute `:requeue` to be true or false here, which will allow us to put less data in the message.